docs/ConvertDataOperation.md

```diff
@@ -0,0 +1,110 @@
+
+| Parameter Name      | Description | Accepted values |
+| ----------- | ----------- | ----------- |
+| inputData      | Data to be converted | A valid JSON String|
+| inputDataType   | Data type of input | ```HL7v2``` |
+| templateCollectionReference | Reference to a template collection. It can be the default templates, or an image on Azure Container Registry that the FHIR server can access. | ```microsofthealth/fhirconverter:default```, \<RegistryServer\>/\<imageName\>:\<imageTag\>, \<RegistryServer\>/\<imageName\>@\<imageDigest\> |
+| rootTemplate | The root template to use while transforming the data. | ```ADT_A01```, ```OML_O21```, ```ORU_R01```, ```VXU_V04``` |
+
+**Sample request:**
+```json
+{
+    "resourceType": "Parameters",
+    "parameter": [
+        {
+            "name": "inputData",
+            "valueString": "MSH|^~\\&|SIMHOSP|SFAC|RAPP|RFAC|20200508131015||ADT^A01|517|T|2.3|||AL||44|ASCII\nEVN|A01|20200508131015|||C005^Whittingham^Sylvia^^^Dr^^^DRNBR^PRSNL^^^ORGDR|\nPID|1|3735064194^^^SIMULATOR MRN^MRN|3735064194^^^SIMULATOR MRN^MRN~2021051528^^^NHSNBR^NHSNMBR||Kinmonth^Joanna^Chelsea^^Ms^^CURRENT||19870624000000|F|||89 Transaction House^Handmaiden Street^Wembley^^FV75 4GJ^GBR^HOME||020 3614 5541^HOME|||||||||C^White - Other^^^||||||||\nPD1|||FAMILY PRACTICE^^12345|\nPV1|1|I|OtherWard^MainRoom^Bed 183^Simulated Hospital^^BED^Main Building^4|28b|||C005^Whittingham^Sylvia^^^Dr^^^DRNBR^PRSNL^^^ORGDR|||CAR|||||||||16094728916771313876^^^^visitid||||||||||||||||||||||ARRIVED|||20200508131015||"
+        },
+        {
+            "name": "inputDataType",
+            "valueString": "Hl7v2"
+        },
+        {
+            "name": "templateCollectionReference",
+            "valueString": "microsofthealth/fhirconverter:default"
+        },
+        {
+            "name": "rootTemplate",
+            "valueString": "ADT_A01"
+        }
+    ]
+}
+```
+
+**Sample response:**
+
+Note the Content-Type is the response header is set to `text/plain` because the output format is determined by the mapping definition from your templates.
+
+```
+{
+  "resourceType": "Bundle",
+  "type": "transaction",
+  "entry": [
+    {
+      "fullUrl": "urn:uuid:9d697ec3-48c3-3e17-db6a-29a1765e22c6",
+      "resource": {
+        "resourceType": "Patient",
+        "id": "9d697ec3-48c3-3e17-db6a-29a1765e22c6",
+        ...
+        ...
+      "request": {
+        "method": "PUT",
+        "url": "Location/50becdb5-ff56-56c6-40a1-6d554dca80f0"
+      }
+    }
+  ]
+}
+```
+
+### 3. Evaluate default templates
+When you deploy the FHIR server, a copy of the latest template released by the [FHIR Converter](https://github.com/microsoft/FHIR-Converter/tree/dotliquid) project is stored with the FHIR server. You can use these default templates by passing ```microsofthealth/fhirconverter:default``` in the request payload as described earlier.
```

Comment:
Are the templates stored if the fhir server isn't created with convert data flag enabled? Does a user need to do extra steps if they enable convert data on an existing server?

Reply:
The default templates are packed into the service code as EmbededResource, it is inside the service regardless convert data flag is on. If a user want to activate convert data on an existing server, he only need to set the flag as true.